src/sync-merkleroots.test.ts

```diff
@@ -0,0 +1,315 @@
+import fetchMock from 'jest-fetch-mock';
+import {
+  ErrorStaleLastEvaluatedKey,
+  ErrorSyncMerkleRootsTimeout,
+  ExclusiveStartKeyPage,
+} from '../dist/typescript-npm-package.cjs';
+import { SpvWalletClient } from './client';
+import { MerkleRoot, MerkleRootsRepository } from './types';
+
+const server = 'http://localhost:3003/api/v1';
+
+const client = new SpvWalletClient(server, {
+  xPriv:
+    'xprv9s21ZrQH143K3axKPtYBDKsrAvN3J85z6nZuW5ihYK8JpDWfqHdjswvUnbXzUpMBne1WD6FQmTzymB4Pt3u3UVSauxzq5PswBWr3vYtowmW',
+});
+
+// creates a simulated repository a client passes to SyncMerkleRoots()
+const createRepository = (dataBase: { merkleRoots: MerkleRoot[] }): MerkleRootsRepository => {
+  return {
+    saveMerkleRoots: (syncedMerkleRoots: MerkleRoot[]) => {
+      console.log('\nsaveMerkleRoots called\n');
+      dataBase.merkleRoots.push(...syncedMerkleRoots);
+
+      return new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
I'm not sure if this `setTimeout` is necessary. If you really want to simulate that some function "works" some time - consider decrease the value to a couple of miliseconds (e.g. `5` 🙈)

Reply:
changed to 5. It is necesary to simulate the timeout otherwise sometimes the test doesn't pass as it completes it faster than it should